File: list/ans28.py
# nums = [1, 2, 3, 4, 4]
# nums = [1, 1, 1, 0, 0, 0, 2, -2, -2]
nums = [2, 2]

buffer = []

for i in range(0, len(nums)):
    if nums[i] not in buffer:
        buffer.append(nums[i])

if len(buffer) <= 1:
    print(None)
else:
    print(buffer,  buffer[-2])

File: list/ans28.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main(){
    std::vector<int> nums = {1, 2, 3, 4, 4};
    std::vector<int> buffer;
    for (int i = 0; i<nums.size(); i++){
        auto  it = std::find(buffer.begin(),buffer.end(),nums[i]);
        if(it == buffer.end()){
            buffer.push_back(nums[i]);
        }
    }

    std::cout << "[";
    for(int i = 0; i<buffer.size();i++){
        std::cout << buffer[i];
        if(i != buffer.size()-1){
            std::cout <<", ";
        }
    }
    std::cout<<"]"<<" "<< buffer[buffer.size()-2]<<std::endl;
}